apps/stats-dapp/src/provider/stats-provider.tsx

```diff
@@ -148,10 +158,21 @@ export const StatsProvider: React.FC<Omit<StatsProvidervalue, 'isReady' | 'metaD
       },
       isReady,
       metaData,
+      api,
     };
-  }, [staticConfig, metaData, isReady, time]);
+  }, [staticConfig, metaData, isReady, time, api]);
   const query = useLastBlockQuery();
 
+  useEffect(() => {
+    getPromiseApi();
+  }, [polkadotProviderEndpoint]);
```

Comment:
When you move the `getPromiseApi` into this `useEffect` hook, you don't need the `polkadotProviderEndpoint` in the dependency array because it has a static value.